sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala

```diff
@@ -91,7 +91,7 @@ case class AddJar(path: String) extends RunnableCommand {
     val jarURL = new java.io.File(path).toURL
     val newClassLoader = new java.net.URLClassLoader(Array(jarURL), currentClassLoader)
     Thread.currentThread.setContextClassLoader(newClassLoader)
-    org.apache.hadoop.hive.ql.metadata.Hive.get().getConf().setClassLoader(newClassLoader)
+    hiveContext.executionHive.state.getConf.setClassLoader(newClassLoader)
```

Comment:
This is very likely to work for almost all cases, but we should probably add a TODO or JIRA to clean up some of this stuff. For instance, here we may be setting the executionHive class loader to an entirely different class loader than it originally had (which is likely to work in typical situations, but is not easy to trace). 

Additionally, the flow of class loader-setting inside ClientWrapper is very non-obvious, especially as it differs between Hive 12 and Hive 13 -- we should explicitly document this if not make it cleaner.